### PR TITLE
[air] add to_air_checkpoint method for inference only workload.

### PR DESCRIPTION
**Pull request by xwjiang2010**
## Why are these changes needed?

Follow up on our last discussion for supporting piecemeal fashion air users.
Only did for tensorflow for now, want to collect some feedback on API naming, package structure etc and I will add others.

## Related issue number

#25224

## Checks

- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [X] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by richardliaw — CHANGES_REQUESTED** [review]
Add to package ref and mention it in key concepts?

File: python/ray/ml/predictors/integrations/tensorflow/utils.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+
+def to_air_checkpoint(
+    model: keras.Model, preprocessor: Optional[Preprocessor] = None
+) -> Checkpoint:
+    """Convert a pretrained model to AIR checkpoint for serve or inference.
+
+    Args:
+        model: A pretrained model.
+        preprocessor: Stateless preprocessor only. The preprocessing logic will
```
[the remark this answers is not in the document]
Comment (xwjiang2010) [reply python/ray/ml/predictors/integrations/tensorflow/utils.py]: Rephrased..

**Comment by xwjiang2010** [comment]
> Add to package ref and mention it in key concepts?

Added a session in key concepts

File: doc/source/ray-air/use-pretrained-model.rst [hunk 2]
```diff
@@ -0,0 +1,21 @@
+.. _use-pretrained-model:
+
+Use pretrained model
+========================
```
Comment (richardliaw) [inline doc/source/ray-air/use-pretrained-model.rst:4]:
```suggestion
Use a pretrained model for batch inference
===============================
```

File: python/ray/air/predictors/integrations/tensorflow/utils.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+from ray.air.constants import MODEL_KEY, PREPROCESSOR_KEY
+from ray.air.preprocessor import Preprocessor
+
+
+def to_air_checkpoint(
+    model: keras.Model, preprocessor: Optional[Preprocessor] = None
+) -> Checkpoint:
+    """Convert a pretrained model to AIR checkpoint for serve or inference.
+
+    Args:
```
Comment (richardliaw) [inline python/ray/air/predictors/integrations/tensorflow/utils.py:15]: add Returns:

File: doc/source/_toc.yml [hunk 4]
```diff
@@ -182,6 +182,7 @@ parts:
               - file: ray-air/deployment
               - file: ray-air/check-ingest
               - file: ray-air/preprocessors
+              - file: ray-air/use-pretrained-model
```
Comment (richardliaw) [inline doc/source/_toc.yml:185]: rebase => move to User Guide

**Review by richardliaw — APPROVED** [review]
Looks good, did not review new models very carefully -- ping when tests pass

**Review by amogkam — APPROVED** [review]

**Pull request merged**